sdlib/d/gc/hpd.d

```diff
@@ -111,6 +111,74 @@ public:
 		return bestIndex;
 	}
 
+	uint shrink(uint index, uint oldPages, uint delta) {
```

Comment:
Is this shrinking the hdp?

Reply:
Also, how does this end up being so much more complex than release?

Reply:
This does the same thing as release expect it doesn't have to change `allocCount`, yet this is larger. How can this be larger when it does less?

Reply:
This either does things is a way that is more complicated than it needs to be or does things that it shouldn't be doing at all. It's obvious from comparing with the release function. (also it would stand to reason that both have common components to share, it is somewhat surprising that they do not).